src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,129 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final double LOAD_FACTOR = 0.75;
+
+    private Node<K, V>[] buckets;
+
+    private int size;
+    private int capacity = DEFAULT_CAPACITY;
+    private double threshold = capacity * LOAD_FACTOR;
+
+    public MyHashMap() {
+        buckets = new Node[DEFAULT_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
+        if (key == null) {
+            if (buckets[0] == null) {
+                buckets[0] = new Node<>(null, value);
+            } else {
+                Node<K, V> current = buckets[0];
+                while (current != null) {
+                    if (current.key == null) {
+                        current.value = value;
+                        return;
+                    }
+                    if (current.next == null) {
+                        break;
```

Comment:
Since you're updating `current` value and check for null in loop exit condition, this check here is unnecessary. Try to avoid using `break`.

Reply:
These lines don't exist in newer version of my code